Detectors/GlobalTrackingWorkflow/tpcinterpolationworkflow/src/TPCResidualWriterSpec.cxx

```diff
@@ -0,0 +1,81 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file  TPCResidualWriterSpec.cxx
+
+#include <vector>
+
+#include "TTree.h"
+
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "SpacePoints/TrackInterpolation.h"
+#include "TPCInterpolationWorkflow/TPCResidualWriterSpec.h"
+
+namespace o2
+{
+namespace tpc
+{
+
+void ResidualWriterTPC::init(InitContext& ic)
+{
+  mOutFileName = ic.options().get<std::string>("residuals-outfile");
+}
+
+void ResidualWriterTPC::run(ProcessingContext& pc)
+{
+  if (mFinished) {
+    return;
+  }
+
+  TFile outf(mOutFileName.c_str(), "recreate");
+  if (outf.IsZombie()) {
+    LOG(FATAL) << "Failed to open output file " << mOutFileName;
+  }
+
+  TTree tree(mTreeName.c_str(), "Tree of TPC residuals and reference tracks");
```

Comment:
I am trying to change the writers in such a way that they can deal with multiple TFs (one per tree entry). To avoid double work, could you please at once make this device to comply with such a policy? See the example of change in 
https://github.com/AliceO2Group/AliceO2/blob/7f4eb42483c0a20ff07845320ac01d22e45a5e09/Detectors/ITSMFT/ITS/workflow/src/ClusterWriterSpec.cxx#L32-L96
(from https://github.com/AliceO2Group/AliceO2/pull/3326/commits/7f4eb42483c0a20ff07845320ac01d22e45a5e09).
Also, I am modifying the readers to account for multiple TF entries in the tree (there the change is more trivial), once example is ready, will ask you to apply it also in this PR.